// @grammyjs/storage-redis/dist/cjs/mod.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RedisAdapter = void 0;
class RedisAdapter {
    /**
     * @constructor
     * @param {opts} Constructor options
     * @param {opts.ttl} ttl - Session time to life in SECONDS.
     * @param {opts.instance} instance - Instance of redis.
     */
    constructor({ instance, ttl }) {
        if (instance) {
            this.redis = instance;
        }
        else {
            throw new Error('You should pass redis instance to constructor.');
        }
        this.ttl = ttl;
    }
    async read(key) {
        const session = await this.redis.get(key);
        if (session === null || session === undefined) {
            return undefined;
        }
        return JSON.parse(session);
    }
    async write(key, value) {
        await this.redis.set(key, JSON.stringify(value));
        if (this.ttl) {
            this.redis.expire(key, this.ttl);
        }
    }
    async delete(key) {
        await this.redis.del(key);
    }
}
exports.RedisAdapter = RedisAdapter;
